### ISSUE TITLE
Nothing happens when clicking on the icon

### ISSUE THREAD

**Issue by Kabouik** [HackeSta/polybar-kdeconnect#17]
I'm not sure how to trigger the rofi menu, should it show up when clicking on the icon? I see no errors when running polybar from terminal and clicking on the icon, but nothing happens.

**Comment by haideralipunjabi** [comment]
It should only show up if the device is connected. What is the color of the icon you are clicking?

**Comment by Kabouik** [comment]
Oh, I see, so I need to add my devices using kdeconnect-cli first I suppose (not sure why, but it's not listed in `kdeconnect-cli -l` despite being on the same network and with kdeconnect on on both ends). I changed the colours in your script but the icon shown is the icon corresponding to no device connected, indeed.

**Comment by haideralipunjabi** [comment]
An unpaired device is shown in Yellow Color (default colours). You can pair using that icon as well. A device has two boolean values associated with it, `isReachable` and `isTrusted`. An unpaired device will have `isReachable = true` and `isTrusted = false`.

**Comment by octal-illumination** [comment]
Hi I am having a similar issue here, and in my case, the device is connected. I have black icons shaped as circle square and triangle when device is paired and not reachable, red colored icons when the device is paired and reachable. events from kdeconnect app is working on either direction, but unfortunately no rofi menu as experienced by the OP. AM i doing something wrong?

![red_threeshapes](https://user-images.githubusercontent.com/87659315/126330706-28cdc0cc-6428-45ba-a5d9-574a14cadbfd.png)

![black_threeshapes](https://user-images.githubusercontent.com/87659315/126330710-59521a25-4af2-4fd1-8417-130bedc0d2a0.png)

My Polybar-kdeconnect config:
```
[module/kdeconnect]
type = custom/script
exec = "~/.config/polybar/scripts/kdeconnect/polybar-kdeconnect.sh -d"
tail = true
```

I can't figure out what I am doing wrong here. Please let me know if you need any other information.

**Comment by haideralipunjabi** [comment]
What happens when you execute the following in terminal:

`rofi -sep "|" -dmenu -i -p DEMO -location 0 <<< "Row1|Row2"`

**Comment by octal-illumination** [comment]
> `rofi -sep "|" -dmenu -i -p DEMO -location 0 <<< "Row1|Row2"`

I get the rofi demo menu with Row1 and Row2

**Comment by octal-illumination** [comment]
Additionally i get grey icons once in a while when the device is paired and reachable, verified when i check with `kdeconnect-cli -l`, i get 
```
- Device name: Device ID (paired and reachable)
1 device found
```
This is the same result i get when i have red icons. 
![grey_threeshapes](https://user-images.githubusercontent.com/87659315/126427598-e2687c68-7433-468f-9d22-65878a1f0bdd.png)

Not sure why different colors for the same state. May be something else is wrong with my set up?

Update: okay, going through the script, i can see it indicates the battery level, which is 79% now. But still no menu or mobile icons though.

**Comment by haideralipunjabi** [comment]
Yeah, the colours change with the battery level of the device. But I still can't figure out why clicking isn't working for you

**Comment by octal-illumination** [comment]
Okay cool. I will try to play around and if i find something, I will post here, thanks

**Comment by ghost** [comment]
I'm having the same issue. Clicking does nothing. Icon is white, and `kdeconnect-cli -l` shows the device as paired and reachable. `rofi -sep "|" -dmenu -i -p DEMO -location 0 <<< "Row1|Row2` brings up the rofi demo menu. 

Has anyone found a solution?

**Comment by k1ake** [comment]
Have the same problem, but i found an issue: script does not assign properly DEV_ID and DEV_NAME, so you have to manually put it inside script or into your polybar config. So i solved that with adding that string in script: DEV_ID="$(kdeconnect-cli -a --id-only)"

**Comment by appcreatorguy** [comment]
Ok, i was having the same issue and found that due to my device name in KDE Connect having single quotes in it, the name getting passed to the script to bring up the rofi menu was being escaped, causing an error.
Changing the device name in KDE Connect settings to remove the single quotes fixed this for me
